### PR TITLE
[VAS] BUG:10888 - fix: internal server error in file formats viewer

### PR DESCRIPTION
**Pull request by Regzox**
## Description

* fix bad api hydration in frontend component common option tab
* fix bad usage of endsWith method in history ui backend controller
* remove puid encoding in external referential api history call
* add tests on get file-formats and history logbooks

## Contributeur

VAS (Vitam Accessible en Service)

### CODE REVIEW

**Review by laedanrex — APPROVED** [review]

File: ui/ui-referential/src/main/java/fr/gouv/vitamui/referential/rest/FileFormatController.java [hunk 1]
```diff
@@ -130,14 +130,39 @@ public PaginatedValuesDto<FileFormatDto> getAllPaginated(@RequestParam final Int
     @RequestMapping(value = "/**", method = RequestMethod.GET)
     @ResponseStatus(HttpStatus.OK)
     public Object getByIdOrHistory(HttpServletRequest request) throws UnsupportedEncodingException, InvalidParseOperationException {
-        LOGGER.debug("getByIdOrHistory ");
-        String requestURL = request.getRequestURL().toString();
-        String path = StringUtils.substringAfter(requestURL,"/fileformat/");
-        if (StringUtils.endsWith("/history", path)) {
-            return findHistoryById(StringUtils.substringBefore(path,"/history"));
+        LOGGER.trace("Entering in 'getByIdOrHistory' method");
```
Comment (oussamasic) [inline ui/ui-referential/src/main/java/fr/gouv/vitamui/referential/rest/FileFormatController.java]: debug ?

File: ui/ui-referential/src/main/java/fr/gouv/vitamui/referential/rest/FileFormatController.java [hunk 2]
```diff
@@ -130,14 +130,39 @@ public PaginatedValuesDto<FileFormatDto> getAllPaginated(@RequestParam final Int
...
+
+        final String requestUrl = request.getRequestURL().toString();
+        final String fileFormatId = extractFileFormatId(requestUrl);
+        final boolean isHistoryRequest = isHistoryRequest(requestUrl);
+
+        LOGGER.debug("Search history for {} file format", fileFormatId);
+
+        if (isHistoryRequest) {
+            return findHistoryById(fileFormatId);
```
Comment (oussamasic) [inline ui/ui-referential/src/main/java/fr/gouv/vitamui/referential/rest/FileFormatController.java]:
```suggestion
return isHistoryRequest ? findHistoryById(fileFormatId) 
: getById(fileFormatId);
```

File: ui/ui-referential/src/main/java/fr/gouv/vitamui/referential/rest/FileFormatController.java (hunk 1 above)
Comment (Regzox) [reply ui/ui-referential/src/main/java/fr/gouv/vitamui/referential/rest/FileFormatController.java]: C'est remplacé

File: ui/ui-referential/src/main/java/fr/gouv/vitamui/referential/rest/FileFormatController.java (hunk 2 above)
Comment (Regzox) [reply ui/ui-referential/src/main/java/fr/gouv/vitamui/referential/rest/FileFormatController.java]: Remplacé par une ternaire.

**Review by oussamasic — APPROVED** [review]

**Review by bbenaissa — APPROVED** [review]

**Pull request merged**